dashboard/app/public_json_api_test.go

```diff
@@ -250,3 +257,78 @@ func TestPublicJSONAPI(t *testing.T) {
 	c.expectOK(err)
 	c.expectEQ(config, []byte("config1"))
 }
+
+func TestWriteExtAPICoverageFor(t *testing.T) {
+	ctx := SetCoverageDBClient(context.Background(), fileFuncLinesDBFixture(t,
+		[]*coveragedb.FuncLines{
+			{
+				FilePath: "/file",
+				FuncName: "func_name",
+				Lines:    []int64{1, 2, 3},
+			},
+		},
+		[]*coveragedb.FileCoverageWithLineInfo{
+			{
+				FileCoverageWithDetails: coveragedb.FileCoverageWithDetails{
+					Filepath: "/file",
+				},
+				LinesInstrumented: []int64{1, 2, 3},
+				HitCounts:         []int64{10, 20, 30},
+			},
+		},
+	))
+
+	var buf bytes.Buffer
+	err := writeExtAPICoverageFor(ctx, &buf, "test-ns", "test-repo")
+	assert.NoError(t, err)
+	assert.Equal(t, `{
+	"repo": "test-repo",
+	"commit": "",
+	"file_path": "/file",
+	"functions": [
+		{
+			"func_name": "func_name",
+			"total_blocks": 3,
+			"covered_blocks": [
+				{
+					"from_line": 1,
+					"from_column": 0,
+					"to_line": 1,
+					"to_column": -1
```

Comment:
Out of curiosity, why are the values of "to_column" -1, given that you don't actually support columns?

Reply:
https://github.com/google/syzkaller/issues/5591#issuecomment-2582148934 this comment and two following to linkify the context.

We'll need the range support in the future (high confidence). The alternative is to omit to_line now, but in the future empty lines will be parsed as 0.